services/src/datasets/external/gbif.rs

```diff
@@ -0,0 +1,2467 @@
+    use crate::layers::layer::Layer;
+    use crate::layers::layer::ProviderLayerCollectionId;
+    use crate::test_data;
+    use crate::util::{config, user_input::UserInput};
+
+    use super::*;
+
+    /// Create a schema with test tables and return the schema name
+    async fn create_test_data(db_config: &config::Postgres) -> String {
+        let mut pg_config = Config::new();
+        pg_config
+            .user(&db_config.user)
+            .password(&db_config.password)
+            .host(&db_config.host)
+            .dbname(&db_config.database);
+        let pg_mgr = PostgresConnectionManager::new(pg_config, NoTls);
+        let conn = pg_mgr.connect().await.unwrap();
+
+        let mut sql = String::new();
+        File::open(test_data!("gbif/test_data.sql"))
+            .unwrap()
+            .read_to_string(&mut sql)
+            .unwrap();
+
+        let schema = format!("geoengine_test_{}", rand::thread_rng().next_u64());
+
+        conn.batch_execute(&format!(
+            "CREATE SCHEMA {schema}; 
+            SET SEARCH_PATH TO {schema}, public;
+            {sql}"
+        ))
+        .await
+        .unwrap();
+
+        schema
+    }
+
+    /// Drop the schema created by `create_test_data`
+    async fn cleanup_test_data(db_config: &config::Postgres, schema: String) {
+        let mut pg_config = Config::new();
+        pg_config
+            .user(&db_config.user)
+            .password(&db_config.password)
+            .host(&db_config.host)
+            .dbname(&db_config.database);
+        let pg_mgr = PostgresConnectionManager::new(pg_config, NoTls);
+        let conn = pg_mgr.connect().await.unwrap();
+
+        conn.batch_execute(&format!("DROP SCHEMA {schema} CASCADE;"))
+            .await
+            .unwrap();
+    }
+
+    #[tokio::test]
+    async fn it_lists_select_items() {
+        let db_config = config::get_config_element::<config::Postgres>().unwrap();
+
+        let test_schema = create_test_data(&db_config).await;
+
+        let provider = Box::new(GbifDataProviderDefinition {
+            name: "GBIF".to_string(),
+            db_config: DatabaseConnectionConfig {
+                host: db_config.host.clone(),
+                port: db_config.port,
+                database: db_config.database.clone(),
+                schema: test_schema.clone(),
+                user: db_config.user.clone(),
+                password: db_config.password.clone(),
+            },
+        })
+        .initialize()
+        .await
+        .unwrap();
+
+        let root_id = provider.root_collection_id().await.unwrap();
+
+        let collection = provider
+            .collection(
+                &root_id,
+                LayerCollectionListOptions {
+                    offset: 0,
+                    limit: 10,
+                }
+                .validated()
+                .unwrap(),
+            )
+            .await;
+
+        cleanup_test_data(&db_config, test_schema).await;
```

Comment:
You don't clean up the test data if one of your previous unwraps panics.

You can look at other Postgres stuff, there should be a helper method that unwinds panics and cleans up stuff.

https://github.com/geo-engine/geoengine/blob/master/services/src/pro/contexts/postgres.rs#L772-L808

Reply:
https://github.com/geo-engine/geoengine/pull/718/commits/5b3b832cf8e75667004313d44c73bc6c95c1c4a6